diffrax/_step_size_controller/constant.py

```diff
@@ -47,15 +52,40 @@ def adapt_step_size(
         args: Args,
         y_error: Y | None,
         error_order: RealScalarLike | None,
-        controller_state: RealScalarLike,
-    ) -> tuple[bool, RealScalarLike, RealScalarLike, bool, RealScalarLike, RESULTS]:
+        controller_state: tuple[IntScalarLike, IntScalarLike, RealScalarLike, RealScalarLike]
+    ) -> tuple[bool, 
+            RealScalarLike,
+            RealScalarLike,
+            bool,
+            tuple[IntScalarLike, IntScalarLike, RealScalarLike, RealScalarLike], 
+            RESULTS
+        ]:
         del t0, y0, y1_candidate, args, y_error, error_order
+        step, num_steps, t0_sim, t1_sim = controller_state
+        step += 1
```

Comment:
Very edge-case but I _think_ during nan-detection then JAX will run itself with numpy arrays instead, in which case this would produce an in-place update.